sos/cleaner/__init__.py

```diff
@@ -656,6 +658,27 @@ def obfuscate_file(self, filename, short_name=None, arc_name=None):
 
         return subs
 
+    def obfuscate_directory_names(self, archive):
```

Comment:
We need to correct symlinks to the just-obfuscated directory names. Reproducer:

    mkdir /var/foo
    date > /var/foo/date.txt
    ln -s /var/foo /var/barlink

Add to e.g. `qpid` plugin collection of `/var/barlink`, and then:

    python3 bin/sos report -o qpid --build --batch --clean --keywords foo,bar

it leaves:

1) `var/barlink` with not obfuscated symlink
2) the symlink is broken:

    # file /var/tmp/sosreport-host0-2021-06-16-daowtyx/var/barlink
    /var/tmp/sosreport-host0-2021-06-16-daowtyx/var/barlink: broken symbolic link to foo
    #

Sadly I am not aware of a simple solution to fix both issues when having a chain of symlinks, though.. (or maybe just dummy "traverse all symlinks in the archive and apply obfuscation to source and destination once" does work well in all cases..?).

Reply:
The symlink name obfuscation should be accounting for the to-be-obfuscated directories:

```python
                # generate the obfuscated name of the link target
                _target_ob = self.obfuscate_string(os.readlink(filename))
                # create the newly obfuscated symlink, pointing to the
                # obfuscated target name, which may not exist just yet, but
                # when the actual file is obfuscated, will be created
                os.symlink(_target_ob, _ob_path)
```

The testing for this was a highly populated `/var/log/containers` with links pointing to `/var/log/pods/` directories that include the hostname as part of the directory name. The symlinks are actually created as broken until `obfuscate_directory_names()` comes through and renames those directories under `/var/log/pods/`, at which point the links become valid.

Do you see this happening for any real collections?

Reply:
Ah, got it. The problem is that dir-level symlinks will not pass `isfile()` like other symlinks, so we were skipping over these types of symlinks entirely.

I have this fixed locally and will push an update shortly.

Reply:
This should also be fixed in the most recent branch update.